### ISSUE TITLE
2020 January Paper Reading

### ISSUE THREAD

**Issue by amyxie361** [amyxie361/blog#6]
2020 0110

Richard:
Text Generation with Exemplar-based Adaptive Decoding
https://arxiv.org/pdf/1904.04428.pdf
任务：给定新闻paragraph 抽取类似新闻的summary作为模版 生成summary
搜索相似新闻的summary：用IR抽取相似paragraph
以前的做法：concat paragraph和template 输入到模型里
本文创新点：每个template生成一个rank1的向量作为生成参数矩阵的生成向量
然后用类似SVDfen jie

Sherry:
Evaluation of Word Vector Representations by Subspace Alignment
http://www.cs.cmu.edu/~ytsvetko/papers/qvec.pdf

**Comment by amyxie361** [comment]
2020 0116


Ruixue | Reformer: The Efficient Transformer |   https://openreview.net/attachment?id=rkgNKkHtvB&name=original_pdf
 -- | -- | --
Richard | Plan-and-Write: Towards better automatic storytelling |   https://arxiv.org/abs/1811.05701
Sherry | Improving Question Generation With to the Point Context |   https://arxiv.org/abs/1910.06036

**Comment by amyxie361** [comment]
2020 0131

#s 
[Experience and lessons from Kaggle QA competitions, with top solutions](https://docs.google.com/presentation/d/1XyZDRLylzUPXJibix-J1i5TvA8hPjyNQmK-MBA-qvcQ/edit#slide=id.p)

Sherry 
[The Reversible Residual Network: Backpropagation Without Storing Activations](https://arxiv.org/pdf/1707.04585.pdf)
reformer用到的不用储存activation也能反传的方法
好处是节省memory，而且可逆网络没有信息损失
缺点是只能做res-net类似的结构，而且中间模块如果不可逆仍然需要存参数

**Comment by amyxie361** [comment]
2020 0207

Sherry
[Learning Neural Templates for Text Generation](https://arxiv.org/pdf/1808.10122.pdf)
任务：table to text
创新点：引入HMM，使得生成可理解 可控制
用conditional hidden semi Markov model  做segment的同时学到模版
用有限个隐状态z表示模版，
z每个step预测一个长度，
长度l和z的embedding输入到RNN里再预测当前的生成
然后loss反传同时更新RNN和HMM的参数[TODO: 怎么同时学HMM的]
最后学到的Z可以看成模版，用不同的Z序列可以生成不同的语句
缺点：Z的标号和模版对应还需要人看，控制部分选择Z的序列的方法并不自然

**Comment by amyxie361** [comment]
2020 0214
Richard [Do Massively Pretrained Language Models Make Better Storytellers?](https://www.aclweb.org/anthology/K19-1079.pdf)
generation的各种evaluation

Sherry [REALM: Retrieval-Augmented Language Model Pre-Training](https://kentonl.com/pub/gltpc.2020.pdf)
用retrieval 帮助pretrain bert用来做QA

Ruixue [Robust Zero-Shot Cross-Domain Slot Filling with Example Values](https://www.aclweb.org/anthology/P19-1547.pdf)
用example做zero shot slot filling

**Comment by amyxie361** [comment]
2020 0221

Ruxiue [Fine-Tuning Pretrained Language Models: Weight Initializations, Data Orders, and Early Stopping](https://arxiv.org/abs/2002.06305v1)

Finetune LM的时候的参数：

不同的random seed作为超参数，去初始化模型/shuffle数据。 random seed 越多，best model越好且看不出收敛

模型早起表现可以预测后期效果，所以可以 Early stop。

Richard [The Evolved Transformer](https://arxiv.org/pdf/1901.11117.pdf)

探究transformer的结构 用autoML找最佳模型。

每次从基础模型修改一点变量，跨栏的方法筛后面训练的种子模型

最佳的结构是：两层transformer，后面一层的connection分两枝
![image](https://user-images.githubusercontent.com/26669286/76109720-0b309100-5fab-11ea-911d-4949e1b0fae1.png)

Sherry [Tree-structured Decoding for Solving Math Word Problems](https://www.aclweb.org/anthology/D19-1241/)
[Blog](https://github.com/amyxie361/Blog/issues/9#issue-565966383)

**Comment by amyxie361** [comment]
2020 0306

Sherry [A Primer in BERTology: What we know about how BERT works](https://arxiv.org/pdf/2002.12327.pdf)
[Blog](https://github.com/amyxie361/Blog/issues/17#issue-572905868)

Richard [DYNAMIC EVALUATION OF TRANSFORMER LANGUAGE MODELS ](https://arxiv.org/pdf/1904.08378.pdf)
dynamic eval 
language model evaluation的时候用当前句子的前面数据再训练一下
![image](https://user-images.githubusercontent.com/26669286/76110297-4f706100-5fac-11ea-8578-72f39e76fec9.png)

**Comment by amyxie361** [comment]
2020 0313

Richard: 
[Towards a Human-like Open-Domain Chatbot](https://arxiv.org/pdf/2001.09977.pdf)
Google meena 对话机器人
给了SSA(Sensibleness and Specificity Average)的human evaluation, 和perplexity是线性关系
用了大量语料和训练做出来的效果
模型是evolved transformer

**Comment by amyxie361** [comment]
2020 0320
Sherry:

[Top-Down Tree-Structured Text Generation](https://arxiv.org/pdf/1808.04865v1.pdf)

BFS做树上enecoding

**Comment by amyxie361** [comment]
2020 0327

Ruixue

[Pre-trained Models for Natural Language Processing: A Survey](https://arxiv.org/pdf/2003.08271v1.pdf)

**Comment by amyxie361** [comment]
2020 0404

Richard
[Tokenizers: How machines read(blog)](https://blog.floydhub.com/tokenization-nlp/?utm_campaign=NLP%20News&utm_medium=email&utm_source=Revue%20newsletter)

**Comment by amyxie361** [comment]
2020 0417

Ruixue
[CROSS-LINGUAL ABILITY OF MULTILINGUAL BERT: A N EMPIRICAL STUDY](https://arxiv.org/pdf/1912.07840.pdf)

Sherry
[BLEURT: Learning Robust Metrics for Text Generation](https://arxiv.org/pdf/2004.04696.pdf)

**Comment by amyxie361** [comment]
2020 0424

#s 
[Sentence-BERT: Sentence Embeddings using Siamese BERT-Networks](https://arxiv.org/abs/1908.10084)

Richard
[STRUCTBERT: INCORPORATING LANGUAGE STRUCTURES INTO PRE-TRAINING FOR DEEP LANGUAGE UNDERSTANDING](https://arxiv.org/abs/1908.04577)

**Comment by amyxie361** [comment]
2020 0501 

Ruixue 
[Open Intent Extraction from Natural Language Interactions](https://dl.acm.org/doi/abs/10.1145/3366423.3380268)

Sherry
[TANDA: Transfer and Adapt Pre-Trained Transformer Models for Answer Sentence Selection](https://arxiv.org/pdf/1911.04118.pdf)

**Comment by amyxie361** [comment]
2020 0508 
Richard 
[Self-Attention with Relative Position Representations](https://arxiv.org/pdf/1803.02155.pdf)

Sherry
[Lite Transformer with Long-Short Range Attention](https://openreview.net/forum?id=ByeMPlHKPH)

**Comment by amyxie361** [comment]
2020 0513

Ruixue
[A MUTUAL INFORMATION MAXIMIZATION PERSPECTIVE OF LANGUAGE REPRESENTATION LEARNING]( https://openreview.net/pdf?id=Syx79eBKwr)
InfoNCE
用mutual information 替换loglikelyhood做LM等pretrain的模型
用一个lower bound替换mutual information
然后把问题转化成两个view之间的MI
一般是输入一个view 输出一个view

Richard 
[Efficient Content-Based Sparse Attention with Routing Transformers](https://arxiv.org/pdf/2003.05997.pdf)
SOTA wiki103ppl, transformer的作者，没发表
方法：用k-means聚类， 只在同一类里面做attention
问题：k是固定的，但是pretrain和finetune token数量严重分布不同

**Comment by amyxie361** [comment]
2020 0522

Sherry
Data-dependent Gaussian Prior Objective for Language Generation
ICLR2020
见ICLR 2020 issue

**Comment by amyxie361** [comment]
2020 0605

 Ruixue 
[Synthesizer: Rethinking Self-Attention in Transformer Models](https://arxiv.org/pdf/2005.00743.pdf)
探究attention masking的作用
1. dense
![image](https://user-images.githubusercontent.com/26669286/85933280-5b599900-b8a3-11ea-842e-b3218cd3b31b.png)
2. random
用random init 矩阵代替attention矩阵 

结论：
效果没有很明显的下降，attention可能没那么重要/有很多冗余参数

Richard
[Longformer: The Long-Document Transformer](https://arxiv.org/pdf/2004.05150.pdf)
用 global + local attention代替fully connect attention来减少计算量
local: sliding window / sparse attention
global: [cls] / query ... task specific

**Comment by amyxie361** [comment]
2020 0612
Ruixue
[The Curious Case of Neural Text Degeneration](https://arxiv.org/abs/1904.09751)
beam search / top k 不好 - 不够diverse
要用top p（累计到一定概率）
可能存在的问题：
实验做得不好，可能调tempretrue的方法也可以增加diversity还保持低perplexity

Richard
transformers结构改进综述
Hierarchical trans: 用来做文本生成的
![image](https://user-images.githubusercontent.com/26669286/85933363-2c8ff280-b8a4-11ea-86f2-0920ec415ca4.png)
相当于某种链接起来
TransXL: chunk & cache
![image](https://user-images.githubusercontent.com/26669286/85933368-48939400-b8a4-11ea-836b-c879cb74acbe.png)

Compressive Trans: 修改TransXL, 把之前的信息都compress
![image](https://user-images.githubusercontent.com/26669286/85933375-56e1b000-b8a4-11ea-9e14-e8b8685e8e3e.png)

sparseTrans: window/ 小矩阵 设计好的窗口 主要做图像
![image](https://user-images.githubusercontent.com/26669286/85933381-66f98f80-b8a4-11ea-8f7e-18096cc47beb.png)

Adaptive sapn Trans: 基于sparse 学习窗口位置

BP-Trans: 二分
![image](https://user-images.githubusercontent.com/26669286/85933400-95776a80-b8a4-11ea-99fc-3ab9882da64d.png)

Longformer：local + global
![image](https://user-images.githubusercontent.com/26669286/85933404-a0ca9600-b8a4-11ea-8c77-464f0eb12c68.png)

Route Trans: 对token做cluster
![image](https://user-images.githubusercontent.com/26669286/85933420-c5267280-b8a4-11ea-8c24-e3def79e1e0e.png)

lite trans: 一半 cnn

reformer: hash local + chunk

**Comment by amyxie361** [comment]
2020 0619

Sherry 
[Recipes for building an open-domain chatbot](https://arxiv.org/pdf/2004.13637.pdf)
Facebook chatbot, TODO for details

**Comment by amyxie361** [comment]
2020 0626
Mohan
[ASK THE RIGHT QUESTIONS: ACTIVE QUESTION REFORMULATION WITH REINFORCEMENT LEARNING](https://arxiv.org/pdf/1705.07830.pdf)
![image](https://user-images.githubusercontent.com/26669286/85933502-affe1380-b8a5-11ea-9a64-c29bad42f3da.png)
用RL的方法训练paraphraser 生成更有利于QA的question

Sherry 
[RESIDUAL ENERGY-BASED MODELS FOR TEXT GENERATION](https://openreview.net/pdf?id=B1l4SgHKDH)
用discriminater 的exp打分加入整句生成的概率重新采样

**Comment by amyxie361** [comment]
2020/07/03 | Richard | PEGASUS: Pre-training with Extracted Gap-sentences for Abstractive Summarization |   | https://arxiv.org/pdf/1912.08777.pdf
  | Ruixue | Linformer: Self-Attention with Linear Complexity |   | https://arxiv.org/pdf/2006.04768.pdf

**Comment by amyxie361** [comment]
2020/07/10 | Sherry | Interesting papers in ACLSpan Selection Pre-training for Question Answering                https://www.aclweb.org/anthology/2020.acl-main.247.pdf        SMART: Robust and Efficient Fine-Tuning for Pre-trained Natural Language Models through Principled Regularized Optimization                https://arxiv.org/abs/1911.03437        Pre-training Is (Almost) All You Need: An Application to Commonsense Reasoning                https://www.aclweb.org/anthology/2020.acl-main.357.pdf        Leveraging Pre-trained Checkpoints for Sequence Generation Tasks                https://www.mitpressjournals.org/doi/pdf/10.1162/tacl_a_00313        Beyond Accuracy: Behavioral Testing of NLP Models with CheckList                https://www.aclweb.org/anthology/2020.acl-main.442.pdf        Best paper |   | https://docs.google.com/presentation/d/19qd38Mn9f5b9hJyRfV4wv-NulYsrE54zW7ac7uXNtZ4/edit?usp=sharing
  | Minghan | OPTIMUS: Organizing Sentences via Pre-trained Modeling of a Latent Space |   | https://arxiv.org/pdf/2004.04092.pdf

**Comment by amyxie361** [comment]
2020/07/17 | Richard | GMAT: Global Memory Augmentation for Transformers |   | https://arxiv.org/pdf/2006.03274

**Comment by amyxie361** [comment]
2020 0724
Mohan
[Neural Syntactic Preordering for Controlled Paraphrase Generation](https://arxiv.org/pdf/2005.02013.pdf)
![image](https://user-images.githubusercontent.com/26669286/88443841-f7fe5080-cde7-11ea-9f08-fa226ae6c64a.png)
![image](https://user-images.githubusercontent.com/26669286/88443845-064c6c80-cde8-11ea-9ba6-49a8ac65a54f.png)
问题：paraphrase，引入parsing tree
方法：
第一步，根据paraphrase parse tree 对应，学到tree结构shuffle的模型
第二步，根据reorder的和原句，生成新句子
特色：用了PARANMT-50M的数据，
方法可能和seq2seq transformer比也没有显著更好

Richard 
[DeBERTa: Decoding-enhanced BERT with Disentangled Attention](https://arxiv.org/pdf/2006.03654.pdf)
![image](https://user-images.githubusercontent.com/26669286/88443925-70fda800-cde8-11ea-8a08-cb9f4d0d344c.png)
改进了transformerxl的position embedding，采用相对位置，去掉了第四项
然后claim说bert pretrain的时候最后几层其实是用来学word embeding的，但finetune的任务其实不要学这个，pretrain-finetune discrapancy就产生了，他们claim了扔掉最后几层finetune会有更好的效果
